src/main/java/ui/components/pickers/LabelPickerUILogic.java

```diff
@@ -59,58 +71,146 @@ private void populatePanes() {
         if (dialog != null) dialog.populatePanes(getExistingLabels(), getNewTopLabels(), bottomLabels, groups);
     }
 
-    public void toggleLabel(String name) {
-        addRemovePossibleLabel("");
-        preProcessAndUpdateTopLabels(name);
-        updateBottomLabels(""); // clears search query, removes faded-out overlay on bottom labels
+    public ArrayList<String> toggleLabel(String name) {
+        removePossibleLabel();
+        ArrayList<String> removedLabels = processAndUpdateTopLabels(name);
+        restoreBottomLabels(); // clears search query, removes faded-out overlay on bottom labels
```

Comment:
See if you can rename `top` and `bottom` to more meaningful names based on what they represent rather than where they are located.